wasmcloud_host/lib/wasmcloud_host/lattice/state_monitor.ex

```diff
@@ -113,9 +116,53 @@ defmodule WasmcloudHost.Lattice.StateMonitor do
     GenServer.call(:state_monitor, :refmaps_query)
   end
 
+  # handle cloud events from HostCore PubSub
   @impl true
-  def handle_info({:lattice_event, event}, state) do
-    state = handle_event(state, event)
+  def handle_info({:lattice_event, raw_event}, state) do
+    case Cloudevents.from_json(raw_event) do
+      {:ok, event} ->
+        process_event(state, event)
+
+      # No-op
+      _ ->
+        Logger.warning("Received event that couldn't be parsed as a Cloudevent, ignoring")
+        state
+    end
+
+    {:noreply, state}
+  end
+
+  # handle cloud events from NATS
+  @impl true
+  def handle_info(
+        {:msg, %{body: body, topic: topic}},
+        state
+      ) do
+    state =
+      if String.starts_with?(topic, "wasmbus.evt.") do
+        case Cloudevents.from_json(body) do
```

Comment:
tiny nit: we could probably clean this up with a with statement like:

```elixir
with {:ok, event} <- Cloudevents.from_json(body),
        {pk, _pid, _prefix} <- WasmcloudHost.Application.first_host() do
        case event ...
 end 
```

Reply:
Oo yeah good call